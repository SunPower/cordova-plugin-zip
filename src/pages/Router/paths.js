export default {
  ROOT: '/',
  ABOUT: '/about',
  LOGIN: '/login',
  SIGNUP: '/sign-up',
  SIGNUP_SERIAL: '/sign-up/serial',
  SIGNUP_PASSWORD: '/sign-up/password',
  SIGNUP_LOADER: '/sign-up/loader',
  HELP_SERIAL: '/help/serial',
  STORAGE: '/storage',
  HISTORY: '/history',
  MENU: '/menu',
  MENU_SOCIAL_SHARE: '/menu/social-share',
  MENU_HELP_CENTER: '/menu/help-center',
  MENU_LEARNING_CENTER: '/menu/learning-center',
  MENU_PARTNER_PROGRAM: '/menu/partner-program',
  MENU_WIFI_SETTINGS: '/menu/wifi-settings',
  MENU_FEEDBACK: '/menu/feedback',
  MENU_BATTERY_SETTINGS: '/menu/battery-settings',
  MENU_NEW_PRODUCTS: '/menu/new-products',
  MENU_ABOUT_WIFI: '/menu/about-wifi',
  MENU_UPDATE_PASSWORD: '/menu/update-password',
  HELP_LIVE_CHAT: '/help/live-chat',
  HELP_FAQ: '/help/faq',
  HELP_DOCUMENTS: '/help/documents',
  HELP_SUBMIT_A_REQUEST: '/help/submit-a-request',
  NOTIFICATIONS: '/notifications',
  FORGOT_PASSWORD: '/forgot-password',
  FORGOT_PASSWORD_SENT: '/forgot-password-sent',
  FORGOT_PASSWORD_FAILED: '/forgot-password-failed',
  RESET_PASSWORD: '/reset-password/:pkg',
  PROFILE: '/profile',
  SETTINGS: '/settings'
}
