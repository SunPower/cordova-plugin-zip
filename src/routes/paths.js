export const paths = {
  PROTECTED: {
    ROOT: '/',
    CREATE_SITE: '/create-site',
    MENU: '/menu',
    MANAGE_FIRMWARES: '/manage-firwares',
    VERSION_INFORMATION: '/version-information',
    GIVE_FEEDBACK: '/give-feedback'
  },
  LOGIN: '/login',
  LOGOUT: '/logout',
  FORGOT_PASSWORD: '/forgot',
  GET_ASSISTANCE: '/assistance'
}
