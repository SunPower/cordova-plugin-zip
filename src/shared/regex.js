export const PASSWORD_REGEXP = /^(?=.*[a-z])(?=.*\d)[A-Za-z\d]{8,}$/
export const EMAIL_REGEXP = /^\S+@\S+\.\S+$/
export const PHONE_REGEXP = /^(1\s?)?((([0-9]{3}))|[0-9]{3})[\s-]?[0-9]{3}[\s-]?[0-9]{4}$/
