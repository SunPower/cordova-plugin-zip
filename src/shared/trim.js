export const trimObject = obj => {
  return Object.entries(obj)
    .map(([k, v]) => ({
      [k]: typeof v === 'string' || v instanceof String ? v.trim() : v
    }))
    .reduce((acc, kv) => ({ ...acc, ...kv }), {})
}
