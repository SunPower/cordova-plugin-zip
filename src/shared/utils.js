export const either = (condition, whenTrue, whenFalse) =>
  condition ? whenTrue : whenFalse
