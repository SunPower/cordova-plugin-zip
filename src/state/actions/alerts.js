import { createAction } from 'redux-act'
import { httpGet } from '../../shared/fetch'

export const ALERTS_FETCH_INIT = createAction('ALERTS_FETCH_INIT')
export const ALERTS_FETCH_SUCCESS = createAction('ALERTS_FETCH_SUCCESS')
export const ALERTS_FETCH_ERROR = createAction('ALERTS_FETCH_ERROR')

let alertsPollTimer
const ALERTS_POLL_INTERVAL = 1000 * 60

export const pollAlerts = () => {
  return (dispatch, getState) => {
    const callback = async () => {
      const state = getState()
      const isLoggedIn = !!state.user.auth.tokenID

      if (alertsPollTimer && !isLoggedIn) {
        clearInterval(alertsPollTimer)
        return
      }

      const addressId = state.user.auth.addressId

      try {
        dispatch(ALERTS_FETCH_INIT())
        const { status, data } = await httpGet(
          `/address/${addressId}/alerts`,
          state
        )
        return status === 200
          ? dispatch(ALERTS_FETCH_SUCCESS(data))
          : dispatch(ALERTS_FETCH_ERROR(data))
      } catch (err) {
        dispatch(ALERTS_FETCH_ERROR(err))
      }
    }

    if (!alertsPollTimer) {
      callback()
      alertsPollTimer = setInterval(callback, ALERTS_POLL_INTERVAL)
    }
  }
}
export const ALERTS_STOP_POLLING_INIT = createAction('ALERTS_STOP_POLLING_INIT')
export const ALERTS_STOP_POLLING_SUCCESS = createAction(
  'ALERTS_STOP_POLLING_SUCCESS'
)

export const stopPollingAlerts = () => {
  return dispatch => {
    dispatch(ALERTS_STOP_POLLING_INIT())
    if (alertsPollTimer) {
      clearInterval(alertsPollTimer)
      dispatch(ALERTS_STOP_POLLING_SUCCESS())
      return
    }
  }
}

export const ALERTS_DISMISS = createAction('ALERTS_DISMISS')
export const ALERTS_SEEN = createAction('ALERTS_SEEN')
