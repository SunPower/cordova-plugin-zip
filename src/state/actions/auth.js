import { createAction } from 'redux-act'
import { getUser } from './user'
import { httpPost, httpDelete, httpGet } from '../../shared/fetch'
import { trimObject } from '../../shared/trim'

export const WORKFLOW_TYPES = {
  SELF: 1,
  THIRD_PARTY: 2,
  INSTALLER: 3
}

export const LOGIN_INIT = createAction('LOGIN_INIT')
export const LOGIN_SUCCESS = createAction('LOGIN_SUCCESS')
export const LOGIN_ERROR = createAction('LOGIN_ERROR')

export const performLogin = values => {
  return async dispatch => {
    try {
      dispatch(LOGIN_INIT())
      const response = await httpPost('/authenticate', values)
      if (response.status !== 200) {
        dispatch(LOGIN_ERROR(response))
        return
      }
      dispatch(LOGIN_SUCCESS(response.data))
      dispatch(getUser())
    } catch (err) {
      dispatch(LOGIN_ERROR(err))
    }
  }
}

export const CREATE_ACCOUNT_INIT = createAction('CREATE_ACCOUNT_INIT')
export const CREATE_ACCOUNT_SUCCESS = createAction('CREATE_ACCOUNT_SUCCESS')
export const CREATE_ACCOUNT_ERROR = createAction('CREATE_ACCOUNT_ERROR')

export const createAccount = values => {
  return async dispatch => {
    try {
      dispatch(CREATE_ACCOUNT_INIT())
      const response = await httpPost('/register/createAccount', {
        EmailAddress: values.email,
        FirstName: values.name.substr(0, values.name.indexOf(' ')),
        LastName: values.name.substr(values.name.indexOf(' ')),
        Password: values.password,
        Phone: values.phoneNumber,
        SerialNumber: values.serial,
        AddressID: values.addressId,
        TermsAgree: values.termsAgree
      })
      return response.status === 200
        ? dispatch(CREATE_ACCOUNT_SUCCESS(response.data))
        : dispatch(CREATE_ACCOUNT_ERROR(response))
    } catch (err) {
      dispatch(CREATE_ACCOUNT_ERROR(err))
    }
  }
}

export const LOGOUT = createAction('LOGOUT')

export const logout = () => {
  return async (dispatch, getState) => {
    const state = getState()
    await httpDelete('/session', state)
    dispatch(LOGOUT())
  }
}

export const sendResetPasswordEmail = async email => {
  return httpGet(`/passwordreset/sendPasswordResetEmail?emailaddress=${email}`)
}

export const validatePackage = async pkg => {
  return httpGet(`/passwordreset/validatePackage?package=${pkg}`)
}

export const RESET_PASSWORD_INIT = createAction('RESET_PASSWORD_INIT')
export const RESET_PASSWORD_SUCCESS = createAction('RESET_PASSWORD_SUCCESS')
export const RESET_PASSWORD_ERROR = createAction('RESET_PASSWORD_ERROR')

export const resetPassword = (values, pkg) => {
  return async dispatch => {
    try {
      dispatch(RESET_PASSWORD_INIT())
      const payload = {
        ConfirmEmailAddress: values.email,
        NewPassword: values.password,
        package: pkg,
        TermsAgree: values.termsAgree
      }
      const response = await httpPost(
        '/passwordreset/resetPassword',
        trimObject(payload)
      )
      if (response.status !== 200) {
        dispatch(RESET_PASSWORD_ERROR(response))
        return
      }
      dispatch(RESET_PASSWORD_SUCCESS())
      dispatch(
        performLogin(
          trimObject({
            username: values.email,
            password: values.password
          })
        )
      )
    } catch (err) {
      dispatch(RESET_PASSWORD_ERROR(err))
    }
  }
}

export const checkEmail = async email => {
  return httpPost(`/register/checkEmail`, {
    EmailAddress: email,
    WorkflowType: WORKFLOW_TYPES.SELF
  })
}

export const validateSerial = async serial => {
  return httpGet(`/register/validateSerial/${serial}`)
}

export const VALIDATE_SESSION_INIT = createAction('VALIDATE_SESSION_INIT')
export const VALIDATE_SESSION_SUCCESS = createAction('VALIDATE_SESSION_SUCCESS')
export const VALIDATE_SESSION_ERROR = createAction('VALIDATE_SESSION_ERROR')

export const validateSession = () => {
  return async (dispatch, getState) => {
    try {
      const state = getState()
      dispatch(VALIDATE_SESSION_INIT())
      const response = await httpGet('/session', state)
      if (response.status !== 200) {
        dispatch(VALIDATE_SESSION_ERROR(response))
        dispatch(LOGOUT())
        return
      }
      dispatch(VALIDATE_SESSION_SUCCESS())
    } catch (err) {
      dispatch(VALIDATE_SESSION_ERROR(err))
    }
  }
}
