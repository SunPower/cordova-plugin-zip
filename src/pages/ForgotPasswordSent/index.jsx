import React from 'react'
import { Redirect } from 'react-router-dom'
import paths from '../Router/paths'
import Logo from '../../components/Logo'
import BackButtonBar from '../../components/BackButtonBar'
import { useI18n } from '../../shared/i18n'
import { sendResetPasswordEmail } from '../../state/actions/auth'

import './ForgotPasswordSent.scss'

function onClick(history, email) {
  return async () => {
    try {
      const { status } = await sendResetPasswordEmail(email)
      if (status === 200) {
        history.push({
          pathname: paths.FORGOT_PASSWORD_SENT,
          state: {
            email,
            isResend: true
          }
        })
      } else {
        history.push({
          pathname: paths.FORGOT_PASSWORD_FAILED,
          state: {
            email
          }
        })
      }
    } catch (err) {
      history.push({
        pathname: paths.FORGOT_PASSWORD_FAILED,
        state: {
          error: err
        }
      })
    }
  }
}

function ForgotPasswordSent({
  isLoggedIn = false,
  location = {},
  history = {}
}) {
  const t = useI18n()
  const email = location.state && location.state.email
  const isResend = location.state && location.state.isResend

  if (isLoggedIn) {
    return <Redirect to={paths.ROOT} />
  }

  return (
    <section className="forgot-password section full-min-height is-flex">
      <BackButtonBar history={history} to={paths.LOGIN} />
      <div className="container mt-50 mb-100">
        <Logo />
        <div className="container is-spaced has-text-centered mt-40 title is-6">
          <h1>{t('CHECK_YOUR_INBOX')}</h1>
        </div>
        <div className="container is-spaced mt-30 has-text-centered subtitle is-6">
          <h2>
            {isResend
              ? t('FORGOT_PASSWORD_RESEND_TEXT', email)
              : t('FORGOT_PASSWORD_SENT_TEXT')}
          </h2>
        </div>
      </div>
      <div className="container is-spaced mt-100 has-text-centered">
        <span className="link" onClick={onClick(history, email)}>
          {t('RESEND_EMAIL')}
        </span>
      </div>
      <div className="container is-flex mt-50"></div>
    </section>
  )
}

export default ForgotPasswordSent
