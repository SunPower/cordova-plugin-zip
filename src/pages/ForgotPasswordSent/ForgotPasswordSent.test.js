import React from 'react'
import { shallow } from 'enzyme'
import { act } from 'react-dom/test-utils'
import ForgotPasswordSent from '.'
import paths from '../Router/paths'
import * as i18n from '../../shared/i18n'
import * as authActions from '../../state/actions/auth'

describe('ForgotPasswordSent page', () => {
  const mockHistory = {
    push: jest.fn()
  }
  const email = 'email@sunpower.com'
  const mockLocation = {
    state: {
      email
    }
  }
  let sendMailSpy
  beforeEach(() => {
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
    sendMailSpy = jest.spyOn(authActions, 'sendResetPasswordEmail')
  })

  it('renders without crashing', () => {
    const component = shallow(<ForgotPasswordSent />)
    expect(component).toMatchSnapshot()
  })

  it('redirects if we are logged in', () => {
    const component = shallow(<ForgotPasswordSent isLoggedIn="true" />)
    expect(component).toMatchSnapshot()
  })

  it('resends the email', async () => {
    sendMailSpy.mockReturnValue({ status: 200 })
    const { component } = mountWithProvider(
      <ForgotPasswordSent history={mockHistory} location={mockLocation} />
    )({})

    await act(async () => {
      component.find('span.link').simulate('click')
    })
    expect(authActions.sendResetPasswordEmail).toBeCalledWith(email)
    expect(mockHistory.push).toBeCalledWith({
      pathname: paths.FORGOT_PASSWORD_SENT,
      state: {
        email,
        isResend: true
      }
    })
  })
})
