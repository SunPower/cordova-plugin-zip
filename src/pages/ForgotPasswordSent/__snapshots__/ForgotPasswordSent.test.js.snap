// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordSent page redirects if we are logged in 1`] = `
<Redirect
  to="/"
/>
`;

exports[`ForgotPasswordSent page renders without crashing 1`] = `
<section
  className="forgot-password section full-min-height is-flex"
>
  <BackButtonBar
    history={Object {}}
    to="/login"
  />
  <div
    className="container mt-50 mb-100"
  >
    <Logo />
    <div
      className="container is-spaced has-text-centered mt-40 title is-6"
    >
      <h1>
        CHECK_YOUR_INBOX
      </h1>
    </div>
    <div
      className="container is-spaced mt-30 has-text-centered subtitle is-6"
    >
      <h2>
        FORGOT_PASSWORD_SENT_TEXT
      </h2>
    </div>
  </div>
  <div
    className="container is-spaced mt-100 has-text-centered"
  >
    <span
      className="link"
      onClick={[Function]}
    >
      RESEND_EMAIL
    </span>
  </div>
  <div
    className="container is-flex mt-50"
  />
</section>
`;
